shipment_management/public/js/get_rates.js

```diff
@@ -1,90 +1,109 @@
-    if (frm.doc.docstatus == 0) {
-        frm.add_custom_button(__('Get Fedex Rates'),
-            function () {
-                _get_fedex_rates(frm)
-            });
-    }
+	if (frm.doc.docstatus == 0) {
+		frm.add_custom_button(__('Get Fedex Rates'), function() {
+			_get_fedex_rates(frm);
+		});
+	}
 }
 
 function _get_fedex_rates(frm) {
-    $.each(frm.doc.taxes, function (index, item) {
-        if (item.account_head == "Freight and Forwarding Charges - JA") {
-            frappe.throw("Shipment Charge has already been added")
-        }
-    })
-    frappe.call({
-        freeze: 1,
-        method: 'shipment_management.api.get_rates_for_doc',
-        args: {
-            doc: frm.doc
-        },
-        callback: function (response) {
-            var options_string = ""
-            var service_dict = {}
-            $.each(response.message, function (index, value) {
-                option = value.label + " - $" + value.fee + "\n"
-                options_string += option
-                service_dict[option.slice(0, -1)] = {
-                    "label": value.label,
-                    "fee": value.fee
-                }
-            })
+	frm.doc.taxes.forEach((item) => {
+		if (item.account_head == "Freight and Forwarding Charges - JA") {
+			frappe.throw("Shipment Charge has already been added");
+		}
+	})
 
-            service_dict["PICK UP - $0"] = {"label" : "PICK UP", "fee" : 0}
-            options_string += "PICK UP - " + "$0" + "\n"
-            frappe.prompt({
-                    "label": "Service Types",
-                    "fieldtype": "Select",
-                    "options": options_string.slice(0, -1),
-                    "reqd": 1
-                },
-                function (data) {
-                    var service_data = service_dict[data.service_types]
-                    if (service_data.fee != 0) {
-                        var row = frappe.model.add_child(frm.doc, "Sales Taxes and Charges", "taxes");
-                        row.charge_type = "Actual";
-                        row.account_head = "Freight and Forwarding Charges - JA";
-                        row.tax_amount = service_data.fee
-                        row.description = "Shipping(" + service_data.label + ")"
+	frappe.call({
+		freeze: 1,
+		method: 'shipment_management.shipengine.api.get_rates',
+		args: {
+			doc: frm.doc,
+			estimate: false
+		},
+		callback: function(response) {
+			let options = [];
+			let service_dict = [];
 
-                    }
-                    frm.set_value("fedex_shipping_method", service_dict[data.service_types].label.replace(/ /g, "_"))
-                    refresh_field("taxes");
-                    frm.save()
-                },
-                "Select Service")
-        }
-    })
+			// form all the available shipping options
+			$.each(response.message, function(index, value) {
+				let estimated_arrival = isNaN(value.estimated_arrival) ? `(${value.estimated_arrival})` : `(${value.days} days)`;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'estimated_arrival' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)